packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -220,20 +237,33 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
 
 export { authMiddleware };
 
+const redirectAdapter = (url: string) => {
+  const res = NextResponse.redirect(url);
+  return setHeader(res, constants.Headers.ClerkRedirectTo, 'true');
+};
```

Comment:
Let's move somewhere in utils and use the redirectAdapter from here: https://github.com/clerk/javascript/blob/84c05e6237bf34d5f14e25ac8bc7059a8e684c1e/packages/nextjs/src/server/clerkMiddleware.ts#L159